### DISCUSSION TITLE
2023 Windows Community Toolkit Feedback

### DISCUSSION THREAD

**Discussion by michael-hawker** [CommunityToolkit/Windows#31] [Announcements]
# [Blog Post](https://devblogs.microsoft.com/ifdef-windows/the-windows-community-toolkit-2023-update/)

See [the blog post here](https://devblogs.microsoft.com/ifdef-windows/the-windows-community-toolkit-2023-update/) for all the information we just announced about this repo and our plans.

## Feedback

We're currently looking for feedback on the following areas:

- #32
- #33  
- #34
- #36 
- #39

## Contributing

If you're looking to hop onboard!

- Try out preview packages from [the main feed](https://dev.azure.com/dotnet/CommunityToolkit/_artifacts/feed/CommunityToolkit-MainLatest):

    https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-MainLatest/nuget/v3/index.json

- Provide general feedback here or in one of the specific topics above. Or share what you've been doing in [Show and Tell](https://github.com/CommunityToolkit/Windows/discussions/categories/show-and-tell).

- Look for an issue related to a component that's been ported or needs porting and add a comment saying you'd like to fix something!

    We'll spin off a sub-task to assign to you to track that work for your PR.

**Comment by mrlacey** [comment]
Congrats team! It's great to see all you've done here and the possibilities it provides going forward.

**Reply by michael-hawker** [reply]
Thanks @mrlacey for your early feedback and work on setting up our initial testing and templating. 🦙❤ We're looking forward to seeing what we can all do with these possibilities now that we can build across these different platforms.

(Also, I just realized I forgot to add the blog link once it was published back above. Fixed!)

**Comment by YegorStepanov** [comment]
What the plans for DataGrid?

**Reply by michael-hawker** [reply]
Great question @YegorStepanov! The DataGrid was originally contributed to the Toolkit as a port of the Silverlight codebase. It was eventually meant to transition into WinUI 2 (see [that issue here](https://github.com/microsoft/microsoft-ui-xaml/issues/1500)).

It's a complex control with nearly 40,000 lines of code and fairly outdated styling and patterns.

@niels9001's noticed many apps just need a few things from it vs. a full-blown cell-based editor, like just displaying columns of static data and allowing uses to resize them, maybe editing a single field here or there, etc...

I recently started exploring in the [Windows Community Toolkit Labs](https://aka.ms/toolkit/labs/windows) the idea of a new paradigm to simplify the complexity of producing this scenario in a few hundred lines of code. While hopefully satisfying 90% of the scenarios we see developers really needing within their apps.

Please check out the [`DataTable` proposal here](https://github.com/CommunityToolkit/Labs-Windows/discussions/415), there's even an initial proof-of-concept PR open you could try. I'll be getting back to this a bit later after we port a few more things over here first from the main toolkit (like #25). Had to ensure we had the updated Sizer controls here too so that I can investigate user-resizable columns. (WCT Labs is also still using 7.x for WCT dependencies, so we need to ensure we have enough stuff ported here first so we can lift it up to vNext and not run into conflicts.)

**Reply by YegorStepanov** [reply]
It looks like this is what I need. Looking forward to trying it! Thanks

**Comment by svsnux** [comment]
There seems to me no movement on the DataTable?  Is someone converting DataGrid?

**Comment by trydalch** [comment]
Would be great to just ad some more samples to the DataTable. Some examples I think would be valuable:
1. Header-based sorting
2. Cell editing 
3. Grid lines
4. Sticky headers
